README.md

```diff
@@ -1,21 +1,148 @@
+
+Parts of P2P editor:
+- Code editor: [Ace](https://github.com/ajaxorg/ace)
+- Database: append only log ([hypercore](https://github.com/mafintosh/hypercore))
+- Communication: WebRTC RTCDataChannel ([webrtc-swarm](https://github.com/mafintosh/webrtc-swarm))
+- Peer discovery: WebRTC signaling server ([signalhub](https://github.com/mafintosh/signalhub))
+- Storage: RAM
+
+Workflow:
+1. Peers are discovered via signaling server.
+2. For each change made in the editor, a delta representing that change is appended to hypercore database.
+3. Hypercore database is constantly replicating between all peers via WebRTC RTCDataChannel.
+4. When a peer receives a delta it is applied to its editor instance.
+
+Data consistency: [Eventual consistency](https://en.wikipedia.org/wiki/Eventual_consistency)
+## 🚀 Environment setup
+
+### 🐳 Docker environment
+
+* Clone this repository: `git clone https://github.com/CodelyTV/p2p-editor`
+* Move to your project folder: `cd p2p-editor`
+* Copy the default environment variables: `cp .env.dist .env`
+* Start the services: `docker-compose up -d`
+* Go to [http://localhost:8000](http://localhost:8000)
+
+### 🎰 Local environment
+
+
+* Clone this repository: `git clone https://github.com/CodelyTV/p2p-editor`
+* Move to your project folder: `cd p2p-editor`
+* Copy the default environment variables: `cp .env.dist .env`
+* Install dependencies: `yarn install` or `npm install`
+* Start dev server: `yarn start` or `npm start`
+* Go to [http://localhost:8000](http://localhost:8000)
+
+
+### ✅ Run the tests
+
+`yarn test` or `npm run test `
+
+## 💻 Live demo
+
+- To start a new live code session open [http://p2p-editor.codely.tv](http://p2p-editor.codely.tv).
+- Share the URL with the users you want to join the session.
+- Start editing the file.
+ 
+That's all! All changes are instantly reflected in the connected peers editor.
+
+Happy P2P coding!
+
+## ℹ️ About
 
-## Architecture
+We started this project in the context of a learning process with the aim of continuing to learn about P2P and put into practice what we learned by building a useful P2P application.
 
-WebRTC
+During this process, all the learning resources that we found and consider useful are published in [awesome-p2p]("https://github.com/CodelyTV/awesome-p2p).
 
-## Live demo
+The initial implementation of P2P editor is based on [Mathias Buus](https://github.com/mafintosh) work. From here thank you very much for sharing so many free (libre) resources and knowledge.
 
-TODO
+P2P editor is a non profits free software supported by user contributions and CodelyTV.
 
-## Contribute
+## 🤝 Contributing
 
-TODO
+How you can contribute?
+- Giving feedback and reporting bugs.
+- Getting involved in development resolving a [good first issue](https://github.com/CodelyTV/p2p-editor/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+- Sharing your knowledge by adding learning resources to [awesome-p2p]("https://github.com/CodelyTV/awesome-p2p).
```

Comment:
Add "reporting bugs" and "suggesting new features" linking to the create new issue page